ceph/ceph_admin/orch.py

```diff
@@ -113,6 +115,48 @@ def check_service(
 
         return False
 
+    def check_service_restart(
+        self,
+        service_name: str,
+        restart_init_time: datetime,
+        timeout: int = 300,
+        interval: int = 5,
+    ) -> bool:
+        """
+        Checks whether the given service was restarted in the previous minute or any other amount of time as specified.
+        Args:
+            service_name (Str): Name of the service
+            restart_init_time (datetime.datetime): Time at which restart was initiated
+            timeout: timeout in seconds
+            interval: interval in seconds
+
+        Returns:
+            True if the service was restarted within the given interval from now, else false
+        """
+        end_time = datetime.now() + timedelta(seconds=timeout)
+
+        while end_time > datetime.now():
+            sleep(interval)
+            out, err = self.ls({"base_cmd_args": {"format": "json"}})
+            out = loads(out)
+            service = [d for d in out if d.get("service_name") == service_name]
+
+            if service:
+                last_refresh = service[0]["status"]["last_refresh"]
+                last_refresh = parser.parse(last_refresh).replace(tzinfo=None)
+                if last_refresh >= restart_init_time:
+                    return True
+                else:
+                    return False
+            else:
+                return False
+            LOG.info(
+                "[%s] check for restart after: %s, retrying"
+                % (service_name, restart_init_time)
+            )
+
+        return False
```

Comment:
These lines of code would not be traversed due to ln:148 to ln:152

Reply:
Rearranged them